crates/fluvio/src/producer/mod.rs

```diff
@@ -190,6 +201,20 @@ impl InnerTopicProducer {
             .push_record(record, partition)
             .await?;
 
+        // Increase record count stat
+        if let Some(client_stats) = &self.client_stats {
```

Comment:
not sure what this does?

Reply:
If we've configured the producer to collect stats, we count each record as we push them into the batch.

Atomically updating the struct through the `ArcSwap` takes a little bit of digging. I can bring this into the `ClientStats` struct clean this up.